### PR TITLE
[18725] Added documentation version fallback

### PR DESCRIPTION
**Pull request by jsantiago-eProsima**
## Description

Currently, documentation builds fail if the current version of Fast DDS does not have a matching tag in ReadTheDocs. This complicated the process of generating testing. 

This PR adds a mechanism to get the current `master` branch from ReadTheDocs should the current version download fail. 

## Contributor Checklist
- [X] Commit messages follow the project guidelines. 
- [NA] The code follows the style guidelines of this project. 
- [NA] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [NA] Any new/modified methods have been properly documented using Doxygen. 
- [NA] Changes are ABI compatible. 
- [NA] Changes are API compatible. 
- [NA] New feature has been added to the `versions.md` file (if applicable).
- [NA] New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [NA] Applicable backports have been included in the description.

## Reviewer Checklist
- [ ] The PR has a milestone assigned.
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -502,12 +502,23 @@ if(BUILD_DOCUMENTATION)
...
         file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/readthedocs_custom_template.cmake [=[
 
-            file(DOWNLOAD "https://fast-dds.docs.eprosima.com/_/downloads/en/v${PROJECT_VERSION}/htmlzip/" "./eprosima-fast-rtps.zip")
+            file(DOWNLOAD "https://fast-dds.docs.eprosima.com/_/downloads/en/v${PROJECT_VERSION}/htmlzip/" "./eprosima-fast-rtps.zip"
+                 STATUS DOWNLOAD_STATUS)
+            list(GET DOWNLOAD_STATUS 0 DOWNLOAD_RC)
+            
+            if (DOWNLOAD_RC)
+                message(STATUS "Unable to download documentation for version ${PROJECT_VERSION}. Falling back to master")
+                file(DOWNLOAD "https://fast-dds.docs.eprosima.com/_/downloads/en/master/htmlzip/" "./eprosima-fast-rtps.zip")
```
Comment (EduPonz) [inline CMakeLists.txt:511]:
```suggestion
                file(DOWNLOAD "https://fast-dds.docs.eprosima.com/_/downloads/en/latest/htmlzip/" "./eprosima-fast-rtps.zip")
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -502,12 +502,23 @@ if(BUILD_DOCUMENTATION)
...
+            endif()
             # TODO: when windows ci CMake version surpasses 17 favor file() instead of UNZIP as in the next line
             # file(ARCHIVE_EXTRACT INPUT "./eprosima-fast-rtps.zip" DESTINATION  "${PROJECT_BINARY_DIR}/doc/")
             execute_process(COMMAND "${UNZIP_EXE}" "./eprosima-fast-rtps.zip" -d "${PROJECT_BINARY_DIR}/doc/")
             file(REMOVE_RECURSE "${PROJECT_BINARY_DIR}/doc/manual")
-            file(RENAME "${PROJECT_BINARY_DIR}/doc/eprosima-fast-rtps-v${PROJECT_VERSION}" "${PROJECT_BINARY_DIR}/doc/manual")
+            if (DOWNLOAD_RC)
+                file(RENAME "${PROJECT_BINARY_DIR}/doc/eprosima-fast-rtps-master" "${PROJECT_BINARY_DIR}/doc/manual")
```
Comment (EduPonz) [inline CMakeLists.txt:518]:
```suggestion
                file(RENAME "${PROJECT_BINARY_DIR}/doc/eprosima-fast-rtps-latest" "${PROJECT_BINARY_DIR}/doc/manual")
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -502,12 +502,23 @@ if(BUILD_DOCUMENTATION)
 
         file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/readthedocs_custom_template.cmake [=[
 
-            file(DOWNLOAD "https://fast-dds.docs.eprosima.com/_/downloads/en/v${PROJECT_VERSION}/htmlzip/" "./eprosima-fast-rtps.zip")
+            file(DOWNLOAD "https://fast-dds.docs.eprosima.com/_/downloads/en/v${PROJECT_VERSION}/htmlzip/" "./eprosima-fast-rtps.zip"
+                 STATUS DOWNLOAD_STATUS)
+            list(GET DOWNLOAD_STATUS 0 DOWNLOAD_RC)
```
Comment (EduPonz) [inline CMakeLists.txt:507]: Maybe add a comment linking to https://cmake.org/cmake/help/latest/command/file.html#transfer and stating that 0 means no error

File: CMakeLists.txt [hunk 4]
```diff
@@ -502,12 +502,23 @@ if(BUILD_DOCUMENTATION)
...
             # TODO: when windows ci CMake version surpasses 17 favor file() instead of UNZIP as in the next line
             # file(ARCHIVE_EXTRACT INPUT "./eprosima-fast-rtps.zip" DESTINATION  "${PROJECT_BINARY_DIR}/doc/")
             execute_process(COMMAND "${UNZIP_EXE}" "./eprosima-fast-rtps.zip" -d "${PROJECT_BINARY_DIR}/doc/")
             file(REMOVE_RECURSE "${PROJECT_BINARY_DIR}/doc/manual")
-            file(RENAME "${PROJECT_BINARY_DIR}/doc/eprosima-fast-rtps-v${PROJECT_VERSION}" "${PROJECT_BINARY_DIR}/doc/manual")
+            if (DOWNLOAD_RC)
+                file(RENAME "${PROJECT_BINARY_DIR}/doc/eprosima-fast-rtps-master" "${PROJECT_BINARY_DIR}/doc/manual")
+            else()
+                file(RENAME "${PROJECT_BINARY_DIR}/doc/eprosima-fast-rtps-v${PROJECT_VERSION}" "${PROJECT_BINARY_DIR}/doc/manual")
+            endif()
```
Comment (EduPonz) [inline CMakeLists.txt:521]: This if-else can be substituted by simple statements if you create a local version var containing either `v${PROJECT_VERSION}` or `latest` depending on the downloaded zip

**Review by EduPonz — CHANGES_REQUESTED** [review]
This looks sound. Once the requested changes are applied, I'll run a Windows installer generation on this branch as CI check

File: CMakeLists.txt (hunk 4 above)
Comment (jsantiago-eProsima) [reply CMakeLists.txt]: I actually tried to implement this but the inline cmake script was not playing nice with the configure step when setting a variable via SET so I left it as is.

**Review by EduPonz — APPROVED** [review]

**Comment by EduPonz** [comment]
Windows installer job here:

* [Build Status](http://jenkins.eprosima.com:8080/view/Fast%20DDS/job/FastRTPS%20Installer%20Windows/449/)

Once it passes, 574f8ff must be reverted

**Review by EduPonz — DISMISSED** [review]

**Comment by jsantiago-eProsima** [comment]
@Mergifyio 2.6.x 2.9.x 2.10.x

**Comment by jsantiago-eProsima** [comment]
@Mergifyio backport 2.6.x 2.9.x 2.10.x

**Pull request merged**